torch/csrc/jit/interpreter.cpp

```diff
@@ -0,0 +1,288 @@
+#include "interpreter.h"
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/generated/aten_dispatch.h"
+#ifdef WITH_CUDA
+#include "torch/csrc/jit/fusion_compiler.h"
+#endif
+
+namespace torch { namespace jit {
```

Comment:
It'd be positively lovely if you could record down which perf optimizations you *know* pushed the needle down, if you remember this from when you were working on the interpreter. (Unless, well, it sprang fully formed from your head as fast as it is today. That's cool too 8)

Reply:
I didn't add optimizations after the fact. It's easier to start with good perf than try to optimize later. Instead there were two goals:

1. Don't allocate memory when you running (there is a single malloc when the Interpreter is created)
2. Scan linearly through memory as much as possible to be friendly to cache.

It is not clear if either of these were necessary but in some sense it is easier writing it this way, then writing it another way and then trying all optimizations.